AirCasting/SessionViews/ABMeasurementsView.swift

```diff
@@ -74,40 +98,9 @@ struct ABMeasurementsView: View {
                 }
             }
         }
-        .onChange(of: isCollapsed, perform: { _ in
+        .onChange(of: isCollapsed, perform: { new in
             if isCollapsed == false && !hasAnyMeasurements {
-                showLoadingIndicator = true
-                sessionDownloader.downloadSessionWithMeasurement(uuid: session.uuid) { result in
-                    switch result {
-                    case .success(let data):
-                        let dataBaseStreams = data.streams.values.map { value in
-                            SynchronizationDataConterter().convertDownloadDataToDatabaseStream(data: value)
-                        }
-                        dataBaseStreams.forEach { stream in
-                            stream.measurements.forEach { measurement in
-                                let location: CLLocationCoordinate2D? = {
-                                    guard let latitude = measurement.latitude,
-                                          let longitude = measurement.longitude else { return CLLocationCoordinate2D(latitude: 200, longitude: 200) }
-                                    return CLLocationCoordinate2D(latitude: latitude, longitude: longitude)
-                                }()
-                                guard let streamID = try? measurementStreamStorage.existingMeasurementStream(session.uuid, name: stream.sensorName) else {
-                                    Log.info("failed to get existing streamID for synced measurements from session \(String(describing: session.name))")
-                                    return }
-                                try? measurementStreamStorage.addMeasurement(Measurement(time: measurement.time, value: measurement.value, location: location), toStreamWithID: streamID)
-                            }
-                            try? measurementStreamStorage.saveThresholdFor(sensorName: stream.sensorName,
-                                                                           thresholdVeryHigh: Int32(stream.thresholdVeryHigh),
-                                                                           thresholdHigh: Int32(stream.thresholdHigh),
-                                                                           thresholdMedium: Int32(stream.thresholdMedium),
-                                                                           thresholdLow: Int32(stream.thresholdLow),
-                                                                           thresholdVeryLow: Int32(stream.thresholdVeryLow))
-                            
-                        }
-                        showLoadingIndicator = false
-                    case .failure(let error):
-                        Log.info("\(error)")
-                    }
-                }
+                measurementsViewModel.syncMeasurements(for: session)
```

Comment:
one of the most urgent things to clean up is usage of `CoreData` inside Views. The proper structure for this should just be:
```Swift
measurementsViewModel.syncMeasurements()
```

and `ViewModel` should have the entity injected on `init`:

```Swift
init(session: SessionEntity, ...) { ... }
```

It's still not perfect as ViewModels should also not talk to CD entities directly, but it's already a huge step forward.
We would also probably still need this entity to be present in the view, but step by step we should remove those dependencies and move it out to viewmodels